File: frontend/src/app/domain/objects/album-object.ts
import {ExternalURLObject} from './external-urlobject';
import {ImageObject} from './image-object';
import {RestrictionObject} from './restriction-object';
import {SimplifiedArtistObject} from './simplified-artist-object';

export interface AlbumObject {
  album_type: string;
  total_tracks: number;
  available_markets: string[];
  external_urls: ExternalURLObject;
  href: string;
  id: string;
  images: ImageObject[];
  name: string;
  release_date: string;
  release_date_precision: string;
  restrictions: RestrictionObject;
  type: string;
  uri: string;
  artists: SimplifiedArtistObject[];
}

File: backend/src/main/java/organizer/backend/domain/objects/AlbumObject.java
package organizer.backend.domain.objects;

public record AlbumObject(String album_type,
                          Integer total_tracks,
                          String[] available_markets,
                          ExternalURLObject external_urls,
                          String href,
                          String id,
                          ImageObject[] images,
                          String name,
                          String release_date,
                          String release_date_precision,
                          RestrictionsObject restrictions,
                          String type,
                          String uri,
                          SimplifiedArtistObject[] artists) {
}
